jupyterhub/singleuser.py

```diff
@@ -22,8 +23,17 @@
 
 try:
     import notebook
+    use_serverapp = False
+    server_package = 'notebook'
+    app_name = 'notebook.notebookapp'
 except ImportError:
-    raise ImportError("JupyterHub single-user server requires notebook >= 4.0")
+    try:
+        import jupyter_server
```

Comment:
Can we make this explicit and based on an env variable (e.g. JUPYTERHUB_SERVER_PACKAGE) rather than try/except? Unconditional try/except makes it impossible to pick when both packages are present.